api/src/main/kotlin/edu/wgu/osmt/richskill/RichSkillController.kt

```diff
@@ -155,63 +142,115 @@ class RichSkillController @Autowired constructor(
+    @GetMapping("${RoutePaths.API}${RoutePaths.LEGACY}${RoutePaths.SKILL_DETAIL}", produces = [MediaType.APPLICATION_JSON_VALUE])
+    @ResponseBody
+    fun legacyByUUID(
+            @PathVariable uuid: String,
+            @AuthenticationPrincipal user: Jwt?
+    ): ApiSkill? {
+        return byUUID(uuid, user)?.let { ApiSkillV2.fromLatest(it, appConfig) }
+    }
+    
+    @RequestMapping("${RoutePaths.API}${RoutePaths.LATEST}${RoutePaths.SKILL_DETAIL}", produces = [MediaType.TEXT_HTML_VALUE])
     fun byUUIDHtmlView(
-        @PathVariable uuid: String,
-        @AuthenticationPrincipal user: Jwt?
+            @PathVariable uuid: String,
+            @AuthenticationPrincipal user: Jwt?
     ): String {
         return richSkillRepository.findByUUID(uuid)?.let {
             if (user == null && it.publishStatus() == PublishStatus.Unarchived) {
                 throw ResponseStatusException(HttpStatus.NOT_FOUND)
             }
-
-            "forward:/skills/$uuid"
+            
+            "forward:/v3/skills/$uuid"
+        } ?: throw ResponseStatusException(HttpStatus.NOT_FOUND)
+    }
+    
+    @RequestMapping(path = [
+        "${RoutePaths.API}${RoutePaths.LEGACY}${RoutePaths.SKILL_DETAIL}",
+        "${RoutePaths.API}${RoutePaths.UNVERSIONED}${RoutePaths.SKILL_DETAIL}"],
+            produces = [MediaType.TEXT_HTML_VALUE])
+    fun legacyByUUIDHtmlView(
+            @PathVariable uuid: String,
+            @AuthenticationPrincipal user: Jwt?
+    ): String {
+        return richSkillRepository.findByUUID(uuid)?.let {
+            if (user == null && it.publishStatus() == PublishStatus.Unarchived) {
+                throw ResponseStatusException(HttpStatus.NOT_FOUND)
+            }
+            
+            "forward:/v2/skills/$uuid"
         } ?: throw ResponseStatusException(HttpStatus.NOT_FOUND)
     }
 
-    @RequestMapping(RoutePaths.SKILL_DETAIL, produces = ["text/csv"])
+    @RequestMapping(path = [
+        "${RoutePaths.VERSIONED_API}${RoutePaths.SKILL_DETAIL}",
+        "${RoutePaths.UNVERSIONED_API}${RoutePaths.SKILL_DETAIL}"
+                           ],
+        produces = ["text/csv"])
     fun byUUIDCsvView(
+        @PathVariable(name = "apiVersion", required = false) apiVersion: String?,
         @PathVariable uuid: String,
         @AuthenticationPrincipal user: Jwt?
     ): HttpEntity<*> {
+        
         return richSkillRepository.findByUUID(uuid)?.let {
             if (user == null && it.publishStatus() == PublishStatus.Unarchived) {
                 throw ResponseStatusException(HttpStatus.NOT_FOUND)
             }
-
             val skill = it.toModel()
             val collections = it.collections.map { it.toModel() }.toSet()
             val result = RichSkillCsvExport(appConfig).toCsv(listOf(RichSkillAndCollections(skill, collections)))
             val responseHeaders = HttpHeaders()
             responseHeaders.add("Content-Type", "text/csv")
+
```

Comment:
revert the random whitespace changes in this file unless you feel they are more stylistically appropriate

Reply:
I am used to leave a line before the return statement...